cmd/clusterctl/phases/pivot.go

```diff
@@ -253,8 +253,10 @@ func moveSecret(from sourceClient, to targetClient, secret *corev1.Secret) error
 	// New objects cannot have a specified resource version. Clear it out.
 	secret.SetResourceVersion("")
 
-	// Remove owner reference.
-	secret.SetOwnerReferences(nil)
+	// remove the UID from ownerReferences as it will be different across clusters
+	for i := 0; i < len(secret.OwnerReferences); i++ {
```

Comment:
So I think this breaks API validation [here](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/validation/objectmeta.go#L74-L76), because an ownerRef can't have an empty UID.